GliaWidgets/Sources/CallVisualizer/ScreenSharing/Model/ScreenSharingViewModel.swift

```diff
@@ -0,0 +1,109 @@
+import SwiftUI
+import Combine
+
+extension CallVisualizer.ScreenSharingView {
+    class Model: ObservableObject {
+        @Published private(set) var orientation: UIInterfaceOrientation
+        let style: ScreenSharingViewStyle
+        let environment: Environment
+        let orientationManager: OrientationManager
+        var delegate: Command<DelegateEvent> = .nop
+        var cancellables: Set<AnyCancellable> = []
+
+        init(style: ScreenSharingViewStyle, environment: Environment) {
+            self.style = style
+            self.environment = environment
+            self.orientationManager = environment.orientationManager
+            self.orientation = orientationManager.orientation
+
+            orientationManager.$orientation
+                .sink { [weak self] orientation in
+                    self?.orientation = orientation
+                }
+                .store(in: &self.cancellables)
+        }
+    }
+}
+
+extension CallVisualizer.ScreenSharingView.Model {
+    func event(_ event: Event) {
+        switch event {
+        case .closeTapped:
+            delegate(.closeTapped)
+        case .endScreenShareTapped:
+            endScreenSharing()
+        }
+    }
+
+    func makeHeaderModel() -> HeaderSwiftUI.Model {
+        let endButtonProps: ActionButtonSwiftUI.Model = .init(
+            style: style.header.endButton,
+            accessibilityIdentifier: "header_end_button",
+            isEnabled: false,
+            isHidden: true
+        )
+
+        var backButton: HeaderButtonSwiftUI.Model?
+        if let endButtonStyle = style.header.backButton {
+            backButton = .init(
+                tap: Cmd(closure: { [weak self] in
```

Comment:
I prefer to use 
```swift 
Cmd { [weak self] in
    self?.delegate(.closeTapped)
},
```

Reply:
fixed [ebd6434](https://github.com/salemove/ios-sdk-widgets/pull/784/commits/ebd643405e6cac2d720b77fa705fbf3b154405b8)